File: C++/DATA STRUCTURE GROUP WORK/src/GroupWork/Pet.cpp
/*
 * Pet.cpp
 *
 *  Created on: 15 Nov 2022
 *      Author: Kyle
 */

#include "Pet.h"

Pet::Pet() {
	this->ticketNumber = 0;
	this->name = " ";
	this->type = "";
	this->age = 0;
}

Pet::Pet(int ticketNumber, string name, string type, int age) {
	this->ticketNumber = ticketNumber;
	this->name = name;
	this->type = type;
	this->age = age;
}

Pet::Pet(Pet &pet) {
	this->ticketNumber = pet.ticketNumber;
	this->name = pet.name;
	this->type = pet.type;
	this->age = pet.age;
}

int Pet::getAge() {
	return age;
}

void Pet::setAge(int age) {
	this->age = age;
}

string& Pet::getName() {
	return name;
}

void Pet::setName(string &name) {
	this->name = name;
}

int Pet::getTicketNumber(){
	return ticketNumber;
}

void Pet::setTicketNumber(int ticketNumber) {
	this->ticketNumber = ticketNumber;
}

string& Pet::getType(){
	return type;
}

void Pet::setType(string &type) {
	this->type = type;
}

void Pet::show() {
	cout << "Pet [ticketNumber=" << ticketNumber << ", name=" << name << ", type=" << type << ", age=" << age << "]"<< endl;
}

File: JAVA/Data Structures Lab/src/groupWork/Pet.java
package groupWork;

public class Pet {
	private int ticketNumber;
	private String name;
	private String type;
	private int age;
	
	public Pet() {
		this.ticketNumber = 0;
		this.name = " ";
		this.type = "";
		this.age = 0;
	}
	
	public Pet(int ticketNumber, String name, String type, int age) {
		this.ticketNumber = ticketNumber;
		this.name = name;
		this.type = type;
		this.age = age;
	}
	
	public Pet(Pet pet) {
		this.ticketNumber = pet.ticketNumber;
		this.name = pet.name;
		this.type = pet.type;
		this.age = pet.age;
	}

	public int getTicketNumber() {
		return ticketNumber;
	}

	public void setTicketNumber(int ticketNumber) {
		this.ticketNumber = ticketNumber;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getType() {
		return type;
	}

	public void setType(String type) {
		this.type = type;
	}

	public int getAge() {
		return age;
	}

	public void setAge(int age) {
		this.age = age;
	}

	public void show() {
		System.out.println(toString());
	}
	
	@Override
	public String toString() {
		return "Pet [ticketNumber=" + ticketNumber + ", name=" + name + ", type=" + type + ", age=" + age + "]";
	}
		
}
